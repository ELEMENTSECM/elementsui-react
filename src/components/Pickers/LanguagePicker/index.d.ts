import * as React from 'react';

export interface LanguagePickerProps {
	/**
	 * HTML id tag of the root element
	 */
	id?: string;
	/**
	 * The list of available 2-letter language codes
	 */
	languageCodes: string[];
	/**
	 * Default language 2-letter language code
	 */
	defaultLanguage?: string;
	/**
	 * Value changed event handler
	 */
	onChange?: (...args: any[]) => any;
}

declare const LanguagePicker: React.SFC<LanguagePickerProps>;

export default LanguagePicker;
