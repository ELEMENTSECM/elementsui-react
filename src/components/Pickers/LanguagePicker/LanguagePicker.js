import * as React from 'react';
import PropTypes from 'prop-types';
import { styles } from './LanguagePicker.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';
import Dropdown from '../../Inputs/Dropdown';
import codes from './LanguagePicker.codes.json';

export function LanguagePicker(props) {
	const { id, languageCodes, defaultLanguage, onChange, styles } = props;
	const classNames = classNamesFunction()(styles, props);

	const dropdownOptions = languageCodes.map(code => {
		return {
			key: code,
			text: (codes[code] && codes[code].name) || code,
			isSelected: code === defaultLanguage
		};
	});

	return (
		<div className={classNames.root}>
			<Dropdown id={id} options={dropdownOptions} onChange={option => onChange(option.key)} />
		</div>
	);
}

LanguagePicker.propTypes = {
	/** HTML id tag of the root element */
	id: PropTypes.string,
	/** The list of available 2-letter language codes */
	languageCodes: PropTypes.arrayOf(PropTypes.string).isRequired,
	/** Default language 2-letter language code */
	defaultLanguage: PropTypes.string,
	/** Value changed event handler */
	onChange: PropTypes.func
};

export default styled(customizable('LanguagePicker', ['theme'])(LanguagePicker), styles);
