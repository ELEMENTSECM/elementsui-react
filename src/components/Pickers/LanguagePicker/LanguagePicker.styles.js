export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-LanguagePicker ec ec-dropdown languagepicker',
			{
				// ---------------------------------------------------
				// Notes on possible css class implementation:
				// ec:elements common component scope
				// ec-dropdown:because its a dropdown component
				// languagepicker:variant of the dropdown component
				// Order:Component scope | Type | Variant
				// In the Doc variant, additional classes can be added
				// ---------------------------------------------------
				// background: props.theme.palette.themePrimary
				// place your styles here
			},
			className
		]
	};
};
