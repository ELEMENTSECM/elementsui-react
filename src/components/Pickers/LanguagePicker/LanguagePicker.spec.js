import * as React from 'react';
import { shallow } from 'enzyme';
import LanguagePicker from './LanguagePicker';

describe('LanguagePicker', () => {
	test('displays correctly', () => {
		const wrapper = shallow(
			<LanguagePicker
				id="testLanguagePicker"
				languageCodes={['en', 'nb', 'nn', 'sv']}
				defaultLanguage="en"
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
