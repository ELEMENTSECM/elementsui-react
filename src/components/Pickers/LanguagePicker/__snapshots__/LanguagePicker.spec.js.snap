// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagePicker displays correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledCustomizedLanguagePicker
    defaultLanguage="en"
    id="testLanguagePicker"
    languageCodes={
      Array [
        "en",
        "nb",
        "nn",
        "sv",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "defaultLanguage": "en",
      "id": "testLanguagePicker",
      "languageCodes": Array [
        "en",
        "nb",
        "nn",
        "sv",
      ],
      "styles": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "defaultLanguage": "en",
        "id": "testLanguagePicker",
        "languageCodes": Array [
          "en",
          "nb",
          "nn",
          "sv",
        ],
        "styles": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
